timelock-impl/src/main/java/com/palantir/atlasdb/timelock/ConjureLockV1Resource.java

```diff
@@ -0,0 +1,123 @@
+import com.palantir.lock.ConjureSimpleHeldLocksToken;
+import com.palantir.lock.LockRefreshToken;
+import com.palantir.lock.LockService;
+import com.palantir.lock.SimpleHeldLocksToken;
+import com.palantir.lock.UndertowConjureLockV1Service;
+import com.palantir.tokens.auth.AuthHeader;
+
+public class ConjureLockV1Resource implements UndertowConjureLockV1Service {
+    private final ConjureResourceExceptionHandler exceptionHandler;
+    private final Function<String, LockService> lockServices;
+
+    private ConjureLockV1Resource(
+            RedirectRetryTargeter redirectRetryTargeter,
+            Function<String, LockService> lockServices) {
+        this.exceptionHandler = new ConjureResourceExceptionHandler(redirectRetryTargeter);
+        this.lockServices = lockServices;
+    }
+
+    public static UndertowService undertow(
+            RedirectRetryTargeter redirectRetryTargeter,
+            Function<String, LockService> lockServices) {
+        return ConjureLockV1ServiceEndpoints.of(new ConjureLockV1Resource(redirectRetryTargeter, lockServices));
+    }
+
+    public static ConjureLockV1ShimService jersey(
+            RedirectRetryTargeter redirectRetryTargeter,
+            Function<String, LockService> lockServices) {
+        return new ConjureLockV1Resource.JerseyAdapter(new ConjureLockV1Resource(redirectRetryTargeter, lockServices));
+    }
+
+    @Override
+    public ListenableFuture<Set<ConjureLockRefreshToken>> refreshLockRefreshTokens(AuthHeader authHeader,
+            String namespace, List<ConjureLockRefreshToken> request) {
+        return exceptionHandler.handleExceptions(() -> {
+            ListenableFuture<Set<LockRefreshToken>> serviceTokens = Futures.immediateFuture(
+                    lockServices.apply(namespace).refreshLockRefreshTokens(getLegacyTokens(request)));
+            return Futures.transform(
+                    serviceTokens,
+                    ConjureLockV1Resource::getConjureTokens,
+                    MoreExecutors.directExecutor());
+        });
+    }
+
+    @Override
+    public ListenableFuture<Boolean> unlockSimple(AuthHeader authHeader, String namespace,
+            ConjureSimpleHeldLocksToken request) {
+        return exceptionHandler.handleExceptions(() -> {
+            SimpleHeldLocksToken serverToken = new SimpleHeldLocksToken(
+                    request.getTokenId(), request.getCreationDateMs());
+            return Futures.immediateFuture(lockServices.apply(namespace).unlockSimple(serverToken));
+        });
+    }
+
+    @VisibleForTesting
+    static List<LockRefreshToken> getLegacyTokens(List<ConjureLockRefreshToken> request) {
+        return request.stream()
+                .map(token -> new LockRefreshToken(token.getTokenId(), token.getExpirationDateMs()))
+                .collect(Collectors.toList());
+    }
+
+    @VisibleForTesting
+    static Set<ConjureLockRefreshToken> getConjureTokens(Set<LockRefreshToken> serverTokens) {
```

Comment:
as above

Reply:
Also, these two methods should just live in a separate utility class and be public imo

Reply:
That's a good call. I will do this refactor